test/compile/mark/bar.test.ts

```diff
@@ -301,6 +301,35 @@ describe('Mark: Bar', () => {
     });
   });
 
+  describe('horizontal binned with offset', () => {
+    const y: PositionFieldDef<string> = {bin: true, field: 'Horsepower', type: 'quantitative'};
+    const model = parseUnitModelWithScaleAndLayoutSize({
+      data: {url: 'data/cars.json'},
+      mark: {type: 'bar', yOffset: 5},
+      encoding: {
+        y,
+        x: {aggregate: 'mean', field: 'Acceleration', type: 'quantitative'}
+      }
+    });
+    const props = bar.encodeEntry(model);
+
+    it('should draw bar with y and y2', () => {
+      expect(props.y2).toEqual([
+        fieldInvalidTestValueRef(y, 'y'),
+        {scale: 'y', field: 'bin_maxbins_10_Horsepower', offset: 5}
+      ]);
+      expect(props.y).toEqual([
+        fieldInvalidTestValueRef(y, 'y'),
+        {
+          scale: 'y',
+          field: 'bin_maxbins_10_Horsepower_end',
+          offset: defaultBarConfig.binSpacing + 5
```

Comment:
Is `yOffset` supposed to only offset the `y` property? Do we also have `y2Offset`?

I'm think about the difference between the following scenarios:
- `yOffset` only on `y`, keep `y2`, change `height`
- `yOffset` on `y`, keep `height`, change `y2`